README.md

```diff
@@ -118,7 +118,113 @@ A signal can contain several actions. For example, if your app subscribes to the
 
 You can define a payload used by all actions, or for a specific action. When you define a method called "payload" this method will be used by all actions that don't have an action specific payload defined. If you wish, you can define an action specific payload by defining a method with the action name followed by `_payload`. As an example, see the create specific payload defined in the `create_payload` method in the example.
 
-Inside the signal class, you will have access to the instance variable `@object_id` which represents the ID of the "object" of the signal (Lesson in this case). You also have access to `object` which will give you the initialized object for that ID. You can customize your signal further as you wish, for example in this signal we've created a private method to alias `object` as `lesson` and using that in our payload definitions.
+Inside the signal class, you will have access to the method `object_id` which represents the ID of the "object" of the signal (Lesson in this case). You also have access to `object` which will give you the initialized object for that ID. You can customize your signal further as you wish, for example in this signal we've created a private method to alias `object` as `lesson` and using that in our payload definitions. You also have access to the `context` method to access any context passed when the signal is sent.
+
+#### Signal Context
+
+Sometimes you'll have context for a signal that is perishable and cannot be retrieved from the database later before sending the signal. The payload methods in the signal are called only when sending a signal and will capture the state of the object at that point; context gives you a way to capture state now in an eager way. Here is an example of a signal using context:
+
+```Ruby
+class LessonSignal < Demux::Signal
+  attributes object_class: Lesson, signal_name: "lesson"
+
```

Comment:
We briefly talked about a future where they may be a DSL that would allow for automatic injection of context into payloads.

```
  # psueodcode
  include_context :generic_context, only: [:destroyed, :created, :other_action]
``

Not necessarily the above as that was just quick example